### PR TITLE
Fix possible race condition on request ctx done #1662

### PR DESCRIPTION
**Pull request by byte0o**
Fix possible race condition on request ctx done #1662

### CODE REVIEW

**Comment by erikdubbelboer** [comment]
Almost all tests are failing.

**Comment by byte0o** [comment]
> Almost all tests are failing.

@erikdubbelboer  
I tested it on this machine and the test failed due to timeout. The code I submitted will not have this problem because the previous pull request was also a test timeout.

**Comment by byte0o** [comment]
> Almost all tests are failing.

@erikdubbelboer  This is caused by the test taking too long. I will check the reason.

**Comment by byte0o** [comment]
> Almost all tests are failing.

@erikdubbelboer Handled, it is a deadlock caused by using lock

File: server.go [hunk 1]
```diff
@@ -2745,7 +2753,7 @@ func (ctx *RequestCtx) Done() <-chan struct{} {
 // RequestCtx.s.done is only closed when the server is shutting down
 func (ctx *RequestCtx) Err() error {
 	select {
-	case <-ctx.s.done:
+	case <-ctx.Done(): // fix  Use unified functions instead of reference variables to converge fetching into one place
```
Comment (erikdubbelboer) [inline server.go:2756]:
```suggestion
	case <-ctx.Done():
```
No need to add this as a comment.

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

File: server.go (hunk 1 above)
Comment (byte0o) [reply server.go]: Already processed

**Comment by erikdubbelboer** [comment]
Thanks!

**Pull request merged**